custom/lib/Auth.php

```diff
@@ -85,25 +86,35 @@ public function createMagicLink(
         MagicLinkWithEmailArgs|MagicLinkWithPhoneArgs|MagicLinkWithUserArgs $args,
         MagicLinkOptions|null $options,
     ): MagicLink {
-        $payload = new CreateMagicLinkRequest();
-        $payload->setType($args->type);
-        $payload->setSend($args->send);
+        $identifier = null;
+        $channel = null;
 
         switch ($args) {
             case $args instanceof MagicLinkWithEmailArgs:
-                $payload->setEmail($args->email);
+                $identifier = $args->email;
+                $channel = MagicLinkChannel::EMAIL;
                 break;
             case $args instanceof MagicLinkWithPhoneArgs:
-                $payload->setPhone($args->phone);
+                $identifier = $args->phone;
+                $channel = MagicLinkChannel::PHONE;
                 break;
             case $args instanceof MagicLinkWithUserArgs:
-                $payload->setUserId($args->userId);
-                $payload->setChannel($args->channel);
+                $identifier = $args->userId;
+                $channel = $args->channel;
                 break;
             default:
                 throw new InvalidArgumentException("args must contain an email, phone, or userId");
         }
 
+        $payload = new CreateMagicLinkRequest(
+            [
+                'identifier' => $identifier,
+                'channel' => $channel,
+                'type' => $args->type,
+                'send' => $args->send,
+            ]
```

Comment:
had to go with a dynamically typed constructor instead of using the more strongly typed setters because the PHP codegen has the channel prop typed as `MagicLinkChannel`, which looks like 
https://github.com/passageidentity/passage-php/blob/5384ee9e1ff6ec4f47ab764c97270dc4b282f846/generated/lib/Model/MagicLinkChannel.php#L41-L48

so `MagicLinkChannel::EMAIL` is actually a string, which throws a type error since `CreateMagicLinkRequest->setChannel` is expecting the class.